### PR TITLE
Feature/wm 133 - Detect git system type

### PR DESCRIPTION
**Pull request by baristaGeek**
## Description
This draft PR allows us to call a different endpoint based on the repo source (github.com, bitbucket.org or gitlab.com).

I want some feedback on the approach pls

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  

## Notes
This is a rough draft. We have these issues to solve first:

1. If done this way, we need to adjust the shapes of the different JSONs to render on our webview
2. For GitLab, we need the project's iid (not the name). How can we get it?

## Acceptance
- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by baristaGeek** [comment]
Currently getting 401 for GitLab and "Token is invalid or not supported for this endpoint." for Bitbucket.

File: src/utils/github/getPRsPerSHAS.ts [hunk 1]
```diff
@@ -7,25 +8,99 @@ export default async function getPRsPerSHAS({
...
+        commitList: shaArray,
+      })
+      .then((res) => res.data)
+      .catch((err) => {
+        let reporter = analyticsReporter();
+        let { message } = err;
+        reporter?.sendTelemetryException(err, { error: message });
+      });
+    issuesItems = issues.items;
+  } else if (repoSource === "gitlab.com") {
```
Comment (EstebanDalelR) [inline src/utils/github/getPRsPerSHAS.ts:36]: probs a switch statement looks better.
We will have enterprise github and private Gitlabs

**Review by EstebanDalelR — COMMENTED** [review]
LGTM for now.
Please extract the functions for fetching

File: src/utils/github/getPRsPerSHAS.ts (hunk 1 above)
Comment (baristaGeek) [reply src/utils/github/getPRsPerSHAS.ts:36]: Not in the short term but I can change the syntax

**Review by EstebanDalelR — APPROVED** [review]

**Pull request merged**